src/invoice_details.rs

```diff
@@ -0,0 +1,185 @@
+use crate::amount::{Amount, ToAmount};
+
+use eel::interfaces::ExchangeRate;
+use eel::{Invoice, InvoiceDescription};
+use std::time::{Duration, SystemTime};
+
+pub struct InvoiceDetails {
+    pub invoice: String,
+    pub amount: Option<Amount>,
+    pub description: String,
+    pub payment_hash: String,
+    pub payee_pub_key: String,
+    pub creation_timestamp: SystemTime,
+    pub expiry_interval: Duration,
+    pub expiry_timestamp: SystemTime,
+}
+
+impl InvoiceDetails {
+    pub(crate) fn from_our_invoice(invoice: Invoice, rate: &Option<ExchangeRate>) -> Self {
```

Comment:
If you prefer "our", feel free to reject the suggestion.
```suggestion
    pub(crate) fn from_locally_issued_invoice(invoice: Invoice, rate: &Option<ExchangeRate>) -> Self {
```